packages/jira-adapter/src/filters/workflowV2/workflow_filter.ts

```diff
@@ -101,41 +137,349 @@ const createWorkflowInstances = async (
+* if there is a modification that removes a status from an active workflow we need status mappings to migrate the issues
+*/
+const filter: FilterCreator = ({ config, client, paginator, fetchQuery, elementsSource }) => {
+  let statusIdToUuid: Record<string, string>
+  const originalChanges: Record<string, InstanceElement> = {}
+  return {
+    name: 'workflowFilter',
+    onFetch: async (elements: Element[]) => {
+      if (!config.fetch.enableNewWorkflowAPI || !fetchQuery.isTypeMatch(JIRA_WORKFLOW_TYPE)) {
+        return { errors: [] }
+      }
+      const jiraWorkflow = findObject(elements, JIRA_WORKFLOW_TYPE)
+      if (jiraWorkflow === undefined) {
+        log.error('JiraWorkflow type was not found')
+        return {
+          errors: [workflowFetchError()],
+        }
+      }
+      setTypeDeploymentAnnotations(jiraWorkflow)
+      await addAnnotationRecursively(jiraWorkflow, CORE_ANNOTATIONS.CREATABLE)
+      await addAnnotationRecursively(jiraWorkflow, CORE_ANNOTATIONS.UPDATABLE)
+      await addAnnotationRecursively(jiraWorkflow, CORE_ANNOTATIONS.DELETABLE)
+      const { workflowIds, errors: fetchWorkflowIdsErrors } = await fetchWorkflowIds(paginator)
+      if (!_.isEmpty(fetchWorkflowIdsErrors)) {
+        return { errors: fetchWorkflowIdsErrors }
+      }
+      const workflowChunks = _.chunk(workflowIds, CHUNK_SIZE)
+      const errors: SaltoError[] = []
+      await awu(workflowChunks).forEach(async chunk => {
+        const { workflowInstances, errors: createWorkflowInstancesErrors } = await createWorkflowInstances(
+          client, chunk, jiraWorkflow
+        )
+        errors.push(...(createWorkflowInstancesErrors ?? []))
+        elements.push(...(workflowInstances ?? []))
+      })
+      return { errors }
+    },
+    preDeploy: async changes => {
+      await awu(changes)
+        .filter(isAdditionOrModificationWorkflowChange)
+        .forEach(async change => {
+          const workflowInstance = getChangeData(change)
+          originalChanges[workflowInstance.elemID.getFullName()] = workflowInstance.clone()
+          const statusInstances = getStatusInstances(change)
+          statusIdToUuid = getUuidMap(statusInstances)
+          const resolvedWorkflowInstance = await resolveValues(workflowInstance, getLookUpName)
+          convertTransitionParametersFields(resolvedWorkflowInstance.value.transitions, convertParametersFieldsToString)
+          walkOnElement({ element: resolvedWorkflowInstance, func: replaceStatusIdWithUuid(statusIdToUuid) })
+          const statusesPayload = getStatusesPayload(statusInstances, statusIdToUuid)
+          workflowInstance.value = getWorkflowPayload(
+            isAdditionChange(change),
+            resolvedWorkflowInstance,
+            statusesPayload,
+          )
+        })
+    },
+    deploy: async changes => {
+      const [relevantChanges, leftoverChanges] = _.partition(
+        changes,
+        change => isAdditionOrModificationWorkflowChange(change)
```

Comment:
nit- this will make `relevantChanges` to be in the right type:
```suggestion
        isAdditionOrModificationWorkflowChange
```